### PR TITLE
plugin types with tests

### LINKED ISSUE #4: Add types and tests with tsd
^^

### PR DESCRIPTION
**Pull request by mikaelkaron**
closes #4

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm, this looks good!

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+export interface JWTOptions extends FastifyJWTOptions {
+  namespace?: string
+  jwks?: boolean | GetJwksOptions
+}
+
+export interface WebhookOptions {
+  url: string | URL | UrlObject
+}
+
+export type CreateSession = (request?: FastifyRequest, reply?: FastifyReply) => Promise<void>
```
Comment (mikaelkaron) [inline index.d.ts:15]: having the `reply` here matches the docs, but I can't find anywhere where it's actually used, only the `request` is used. I think we should change this to only have the request, and update the doc.

Comment (mcollina) [reply index.d.ts:15]: most APIs in Fastify requires both, I tend to have them both to avoid issues in the future and breaking changes

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**